torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -318,55 +291,151 @@ class AttributePropagator {
     }
   }
 
-  // cleanupFrozenModule function cleans up the Frozen module. it performs the
+  bool moduleEscapes(Module& subModule, std::shared_ptr<Graph>& graph) {
+    for (auto& output : graph->outputs()) {
+      if (output->type() == subModule.type()) {
```

Comment:
This is checking pointer equality, I believe you want `*output->type() == *subModule.type()`, or `subModule.type()->isSubtypeOf(output->type())`